### PR TITLE
Improve error messages for concurrent CDK

### PR DESCRIPTION
**Pull request by maxi297**
## What
Adding documentation through comments and improving error messages

### CODE REVIEW

File: airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/thread_pool_manager.py [hunk 1]
```diff
@@ -64,7 +64,11 @@ def _prune_futures(self, futures: List[Future[Any]]) -> None:
                     if optional_exception:
                         # Exception handling should be done in the main thread. Hence, we only store the exception and expect the main
                         # thread to call raise_if_exception
-                        self._most_recently_seen_exception = RuntimeError(f"Failed reading with error: {optional_exception}")
+                        # We do not expect this error to happen. The futures created during concurrent syncs should catch the exception and
+                        # push it to the queue. If this exception occurs, please review the futures and how they handle exceptions.
+                        self._most_recently_seen_exception = RuntimeError(
+                            f"Failed processing a future: {optional_exception}. Please contact the Airbyte team."
```
Comment (natikgadzhi) [inline airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/thread_pool_manager.py:70]: Let's clarify how the users should contact our team. If this situation is a bug and/or requires work from the CDK team, we should ask folks to file an issue on airbytehq/airbyte, right?

**Review by natikgadzhi — COMMENTED** [review]
I think this is definitely an improvement, and as such can be merged — I suggest that we figure out how we want folks to contact us, and state that more specifically. 

I think issues are fine as we will get a triage process in place. The likelihood that we will be able to help in GH issues is higher than on Slack. Right?

File: airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/thread_pool_manager.py (hunk 1 above)
Comment (maxi297) [reply airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/thread_pool_manager.py:70]: I can't confirm as I'm not sure what is our official support process. Maybe @girarda can chime in?

File: airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py [hunk 2]
```diff
@@ -153,9 +153,19 @@ def _add_slice_to_state(self, partition: Partition) -> None:
...
+                # If we track state value using records cursor field, we can only do that if there is one partition. This is because we save
+                # the state every time we close a partition. We assume that if there are multiple slices, they need to be providing
+                # boundaries. There are cases where partitions could not have boundaries:
+                # * The cursor should be per-partition
+                # * The stream state is actually the parent stream state
+                # There might be other cases not listed above. Those are not supported today hence the stream should not use this cursor for
+                # state management. For the specific user that was affected with this issue, we need to:
+                # * Fix state tracking (which is currently broken)
+                # * Make the new version available
+                # * (Probably) ask the user to reset the stream to avoid data loss
```
Comment (clnoll) [inline airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py:165]: Thank you for the details here!

**Review by clnoll — APPROVED** [review]

**Pull request merged**